internal/common/common.go

```diff
@@ -41,50 +42,77 @@ func init() {
 	}
 }
 
-func GetVaultAuthAndTarget(ctx context.Context, c client.Client, obj client.Object) (*secretsv1beta1.VaultAuth, types.NamespacedName, error) {
-	var authRef string
+func GetAuthAndTargetNamespacedName(obj client.Object) (types.NamespacedName, types.NamespacedName, error) {
+	var authRef types.NamespacedName
 	var target types.NamespacedName
-	switch o := obj.(type) {
-	case *secretsv1beta1.VaultPKISecret:
-		authRef = o.Spec.VaultAuthRef
-		target = types.NamespacedName{
-			Namespace: o.Namespace,
-			Name:      o.Name,
-		}
-	case *secretsv1beta1.VaultStaticSecret:
-		authRef = o.Spec.VaultAuthRef
-		target = types.NamespacedName{
-			Namespace: o.Namespace,
-			Name:      o.Name,
-		}
-	case *secretsv1beta1.VaultDynamicSecret:
-		authRef = o.Spec.VaultAuthRef
-		target = types.NamespacedName{
-			Namespace: o.Namespace,
-			Name:      o.Name,
-		}
-	default:
-		return nil, types.NamespacedName{}, fmt.Errorf("unsupported type %T", o)
+
+	m, err := NewSyncableSecretMetaData(obj)
+	if err != nil {
+		return types.NamespacedName{}, types.NamespacedName{}, fmt.Errorf("unsupported type %T", obj)
 	}
 
-	var authName types.NamespacedName
-	if authRef == "" {
+	if m.AuthRef != "" {
+		// todo: some sorta sanity checking?
+		names := strings.Split(m.AuthRef, "/")
+		if len(names) > 1 {
+			authRef.Namespace = names[0]
+			authRef.Name = names[1]
+		} else {
+			authRef.Namespace = obj.GetNamespace()
+			authRef.Name = names[0]
+		}
+	} else {
 		// if no authRef configured we try and grab the 'default' from the
 		// Operator's current namespace.
-		authName = types.NamespacedName{
+		authRef = types.NamespacedName{
 			Namespace: OperatorNamespace,
 			Name:      consts.NameDefault,
 		}
-	} else {
-		authName = types.NamespacedName{
-			Namespace: target.Namespace,
-			Name:      authRef,
+	}
+	target = types.NamespacedName{
+		Namespace: obj.GetNamespace(),
+		Name:      obj.GetName(),
+	}
+	return authRef, target, nil
+}
+
+// According to the spec for secretsv1beta1.VaultAuth, Spec.AllowedNamespaces behave as follows:
+// AllowedNamespaces:
+//
+//	 nil - all namespaces are allowed
+//	[]{}  - no namespaces are allowed
+//	[]{"*"} - with length of 1, all namespaces are allowed
+//	[]{"a","b"} - explicitly namespaces a, b are allowed
+func isValidTargetNamespace(auth *secretsv1beta1.VaultAuth, name types.NamespacedName) bool {
```

Comment:
nit: this seems to check the namespace is allowed, maybe consider renaming to something `isNamespaceAllowed` ?

Reply:
Renamed to `AllowedNamespace`
Fixed in a52da2d